pgml-extension/pgml_rust/src/api.rs

```diff
@@ -230,10 +238,7 @@ fn load_dataset(
     limit: Option<default!(i64, "NULL")>,
 ) -> impl std::iter::Iterator<Item = (name!(table_name, String), name!(rows, i64))> {
     // cast limit since pgx doesn't support usize
-    let limit: Option<usize> = match limit {
-        Some(limit) => Some(limit.try_into().unwrap()),
-        None => None,
-    };
+    let limit: Option<usize> = limit.map(|limit| limit.try_into().unwrap());
```

Comment:
Weird.

Reply:
yeah, turns out you can map things like this...